File: sorting algorithms/bucket_sort.c
#include<stdio.h>
#define MAX 30
int main(){
	int arr[MAX] ;
	int size ;
	printf("%s\n", Enter size : );
	scanf("%d" , &size);
	for(int i=0 ; i<size ; i++){
		scanf("%d" , &arr[i])
	}
	int ul = 50 , ll = -20 , range = ul - ll;
	int count[range] = {0};

	for(int i =0 ; i<size ; i++){

		count[arr[i]-ll]++;

	}

	int j =0;

	for(int i = 0 ; i < range ; i++){

		while(count[i]!=0){
			arr[j] = i + ll;
			j++;
			count[i]--;
		}

	}

	for (int i = 0; i < count; ++i){

		printf("%s\n", arr[i] );
		/* code */
	}





}

File: sorting algorithms/bucket_sort.java
import java.util.*;

class BucketSort{

	static void bucketSort(int []arr , int upRange , int lowerRange){

		int range = upRange - lowerRange + 1;
		int count[] = new int[range];
		int j = 0;

		for(int i=0 ; i<arr.length ; i++){

			count[arr[i]-lowerRange]++;

		}

		for(int i = 0 ; i<range ; i++){

			while(count[i]!=0){
				arr[j] = i + lowerRange;
				j++;
				count[i]--;
			}
 
		}

	}



	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		System.out.println("Enter the size of array :");
		int size = sc.nextInt();
		int []arr = new int[size];
		for(int i=0;i<size;i++){
			arr[i] = sc.nextInt();
		}
		bucketSort(arr , 50 , -20 );

		for(int i=0 ; i<size ; i++){
			System.out.print(arr[i]+" ");
		}


	}
}
